### ISSUE TITLE
Cross-platform reproducibility

### ISSUE THREAD

**Issue by jemus42** [PlantedML/randomPlantedForest#36]
I assumed the following result would be static across platforms due to the fixed seed:

``` r
library(randomPlantedForest)
train <-  mtcars[1:20, ]
test <-  mtcars[21:32, ]

set.seed(23)
rpfit <- rpf(mpg ~., data = train, max_interaction = 3)
pred <- predict(rpfit, test)

mse <- rpfit$fit$MSE(as.matrix(pred$.pred), as.matrix(test$mpg))
mse
#> [1] 17.19055
```

<sup>Created on 2022-08-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

...but according to GitHub actions on all non-macOS platforms, that is [not the case](https://github.com/PlantedML/randomPlantedForest/runs/8089300521?check_suite_focus=true#step:6:157) and now I'm wondering if it's macOS being weird or something with the RNG seeding not going the way we expect, _or_ if there's a random component in the process that's overlooked.

**Comment by jemus42** [comment]
Might be a compiler thing and just `g++` vs `clang`. Have to check what `clang` says on Linux - easy to test and if results are at least "only" compiler-dependant than that shouldn't be too much of an issue. 
Main thing is identifying the source of the inconsistency.

**Comment by jemus42** [comment]
Seems to have been resolved with #32 along the way somehow, at least

``` r
library(randomPlantedForest)
train <-  mtcars[1:20, ]
test <-  mtcars[21:32, ]

ntrees <- 35
set.seed(23)
rpfit <- rpf(mpg ~., data = train, max_interaction = 3, ntrees = ntrees)
pred <- predict(rpfit, test)

mse <- rpfit$fit$MSE(as.matrix(pred$.pred), as.matrix(test$mpg))
mse
#> [1] 14.86974
```

<sup>Created on 2022-08-31 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

gives the same result on macOS with `clang++` and Ubuntu with both `g++` and `clang++`.
Can"t check proper `g++` on macOS because there `g++` is a `clang++` symlink but welp, not important I guess.
